src/app/zap-templates/templates/app/attribute-size-src.zapt

```diff
@@ -0,0 +1,134 @@
+{{> header}}
+
+#include <app/util/af.h>
+#include <app/util/basic-types.h>
+#include <support/SafeInt.h>
+#include <support/logging/CHIPLogging.h>
+
+#include "gen/af-structs.h"
+
+using namespace chip;
+
+// The first 2 bytes specify the number of entries. A value of 0xFFFF means the list in invalid
+// and data is undefined.
+constexpr uint16_t kSizeLengthInBytes = 2u;
+
+void copyListMember(uint8_t * dest, uint8_t * src, bool write, uint16_t * offset, uint16_t length)
+{
+    if (write)
+    {
+        memmove(dest + *offset, src, length);
+    }
+    else
+    {
+        memmove(dest, src + *offset, length);
+    }
+
+    *offset = static_cast<uint16_t>(*offset + length);
+}
+
+uint16_t emberAfCopyList(ClusterId clusterId, EmberAfAttributeMetadata * am, bool write, uint8_t * dest, uint8_t * src, int32_t index)
+{
+    if (index == -1)
+    {
+        memmove(dest, src, am->size);
+        return am->size;
+    }
+
+    if (index == 0)
```

Comment:
this special-case for 0 was not documented at the function declaration.  What is it doing?  Is the indexing 1-based or 0-based?

Reply:
The first 2 bytes of the attribute value contains the actual number of elements in the list. The other bytes hold the actual content.

Reply:
OK, I understand that, but can we make the documentation clear?  It's weird to have "index 0" mean "length of list" and "index 1" mean "first element of the list"....  That meaning of index needs to be really clearly documented.